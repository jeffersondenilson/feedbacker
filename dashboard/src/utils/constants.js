export default {
  LABELS: {
    all: 'Todos',
    issue: 'Problemas',
    idea: 'Ideias',
    other: 'Outros'
  },
  COLORS: {
    all: 'brand-info',
    issue: 'brand-danger',
    idea: 'brand-warning',
    other: 'brand-graydark'
  }
}
