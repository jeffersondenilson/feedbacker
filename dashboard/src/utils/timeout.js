// TODO: teste timeout
export function wait (timeMs) {
  return new Promise(resolve => {
    setTimeout(resolve, timeMs)
  })
}
