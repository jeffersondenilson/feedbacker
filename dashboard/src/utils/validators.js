export function validateEmptyAndLength6 (value) {
  if (!value) {
    return '*Este campo é obrigatório'
  }

  if (value.length < 6) {
    return '*No mínimo 6 caracteres'
  }

  return true
}

export function validateEmptyAndEmail (value) {
  if (!value) {
    return '*Este campo é obrigatório'
  }

  const isValid = /^[a-z0-9._-]+@[a-z0-9]+\.[a-z]+(\.[a-z]+)?$/i.test(value)
  if (!isValid) {
    return '*Este campo precisa ser um e-mail válido'
  }

  return true
}
