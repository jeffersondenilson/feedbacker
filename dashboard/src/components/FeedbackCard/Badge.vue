<template>
  <span
    :class="`bg-${classColor}`"
    class="p-2 text-xs font-black text-white uppercase rounded-full"
    data-test="badge"
  >
    {{ label }}
  </span>
</template>

<script>
import { computed } from 'vue'
import { LABELS, COLORS } from '@/utils/constants'

export default {
  props: {
    type: { type: String, required: true }
  },
  setup (props) {
    const label = computed(() => {
      return LABELS[props.type.toLowerCase()] ?? LABELS.other
    })

    const classColor = computed(() => {
      return COLORS[props.type.toLowerCase()] ?? COLORS.other
    })

    return {
      label,
      classColor
    }
  }
}
</script>
