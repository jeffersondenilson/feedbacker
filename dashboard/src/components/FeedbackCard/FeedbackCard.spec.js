import { shallowMount } from '@vue/test-utils'
import FeedbackCard from '.'
import { wait } from '@/utils/timeout'

// TODO: fazer mock de diff time, se nÃ£o vai falhar todo dia
// ou mudar createdAt do feedback

const feedback = {
  text: 'Muito bom!',
  fingerprint: '490135491',
  id: 'eab759f8-f238-4ff9-ae91-ee1558982329',
  apiKey: 'fcd5015c-10d3-4e9c-b395-ec7ed8850165',
  type: 'OTHER',
  device: 'Chrome 85.0, macOS 10.14',
  page: 'https://feedbacker.com/pricing',
  createdAt: new Date('2020-11-13').getTime()
}

jest.mock('@/utils/timeout')

const wrapperFactory = (props) => {
  return shallowMount(FeedbackCard, {
    props: {
      feedback,
      ...props
    }
  })
}

describe('<FeedbackCard />', () => {
  it('should render closed card', async () => {
    const wrapper = wrapperFactory()

    const details = wrapper.find('[data-test=details]')
    const closedIcon = wrapper.find('[data-test=closed-icon]')

    expect(details.exists()).toBe(false)
    expect(closedIcon.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render open card', async () => {
    const wrapper = wrapperFactory({ isOpened: true })

    const details = wrapper.find('[data-test=details]')
    const closedIcon = wrapper.find('[data-test=closed-icon]')

    expect(details.exists()).toBe(true)
    expect(closedIcon.exists()).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should open card when clicked', async () => {
    const wrapper = wrapperFactory()

    await wrapper.find('[data-test=toggle-card]').trigger('click')

    const details = wrapper.find('[data-test=details]')
    const closedIcon = wrapper.find('[data-test=closed-icon]')
    expect(wrapper.emitted()).toHaveProperty('click')
    expect(details.exists()).toBe(true)
    expect(closedIcon.exists()).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
