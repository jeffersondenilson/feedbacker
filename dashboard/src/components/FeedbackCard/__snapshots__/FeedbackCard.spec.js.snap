// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackCard /> should open card when clicked 1`] = `
<div class="flex flex-col px-8 py-6 rounded cursor-pointer bg-brand-gray" data-test="toggle-card">
  <div class="flex items-center justify-between w-full mb-8">
    <badge-stub type="OTHER"></badge-stub><span class="font-regular text-brand-graydark">171 dias atrás</span>
  </div>
  <div class="text-lg font-medium text-gray-800">Muito bom!</div>
  <div class="flex flex-col lg:flex-row mt-8 animate__animated animate__fadeInUp animate__faster" data-test="details">
    <div class="flex flex-col w-full lg:w-1/2">
      <div class="flex flex-col"><span class="font-bold text-gray-400 uppercase select-none"> Página </span><span class="break-words font-medium text-gray-700">https://feedbacker.com/pricing</span></div>
      <div class="flex flex-col"><span class="font-bold text-gray-400 uppercase select-none"> Dispositivo </span><span class="font-medium text-gray-700">Chrome 85.0, macOS 10.14</span></div>
    </div>
    <div class="flex flex-col w-full lg:w-1/2 lg:ml-5">
      <div class="flex flex-col"><span class="font-bold text-gray-400 uppercase select-none"> Usuário </span><span class="font-medium text-gray-700">490135491</span></div>
    </div>
  </div>
  <!--v-if-->
</div>
`;

exports[`<FeedbackCard /> should render closed card 1`] = `
<div class="flex flex-col px-8 py-6 rounded cursor-pointer bg-brand-gray" data-test="toggle-card">
  <div class="flex items-center justify-between w-full mb-8">
    <badge-stub type="OTHER"></badge-stub><span class="font-regular text-brand-graydark">171 dias atrás</span>
  </div>
  <div class="text-lg font-medium text-gray-800">Muito bom!</div>
  <!--v-if-->
  <div class="flex justify-end mt-8" data-test="closed-icon">
    <icon-stub name="chevron-down" size="24" color="#C0BCB0"></icon-stub>
  </div>
</div>
`;

exports[`<FeedbackCard /> should render open card 1`] = `
<div class="flex flex-col px-8 py-6 rounded cursor-pointer bg-brand-gray" data-test="toggle-card">
  <div class="flex items-center justify-between w-full mb-8">
    <badge-stub type="OTHER"></badge-stub><span class="font-regular text-brand-graydark">171 dias atrás</span>
  </div>
  <div class="text-lg font-medium text-gray-800">Muito bom!</div>
  <div class="flex flex-col lg:flex-row mt-8 animate__animated animate__fadeInUp animate__faster" data-test="details">
    <div class="flex flex-col w-full lg:w-1/2">
      <div class="flex flex-col"><span class="font-bold text-gray-400 uppercase select-none"> Página </span><span class="break-words font-medium text-gray-700">https://feedbacker.com/pricing</span></div>
      <div class="flex flex-col"><span class="font-bold text-gray-400 uppercase select-none"> Dispositivo </span><span class="font-medium text-gray-700">Chrome 85.0, macOS 10.14</span></div>
    </div>
    <div class="flex flex-col w-full lg:w-1/2 lg:ml-5">
      <div class="flex flex-col"><span class="font-bold text-gray-400 uppercase select-none"> Usuário </span><span class="font-medium text-gray-700">490135491</span></div>
    </div>
  </div>
  <!--v-if-->
</div>
`;
